x-pack/plugins/ml/public/explorer/explorer_controller.js

```diff
@@ -949,22 +987,54 @@ module.controller('MlExplorerController', function (
     });
   }
 
+  function updateExplorer() {
+    const cellData = $scope.cellData;
+
+    const jobIds = (cellData !== undefined && cellData.fieldName === VIEW_BY_JOB_LABEL) ? cellData.lanes : $scope.getSelectedJobIds();
+    const timerange = getSelectionTimeRange(cellData);
+    const influencers = getSelectionInfluencers(cellData);
+
+    // The following is to avoid running into a race condition where loading a swimlane selection from URL/AppState
+    // would fail because the Explorer Charts Container's directive wasn't linked yet and not being subscribed
+    // to the anomalyDataChange listener used in loadDataForCharts().
+    function finish() {
+      if ($scope.overallSwimlaneData !== undefined) {
+        mlExplorerDashboardService.swimlaneDataChange.changed(mapScopeToSwimlaneProps(SWIMLANE_TYPE.OVERALL));
+      }
+      if ($scope.viewBySwimlaneData !== undefined) {
+        mlExplorerDashboardService.swimlaneDataChange.changed(mapScopeToSwimlaneProps(SWIMLANE_TYPE.VIEW_BY));
+      }
+      mlExplorerDashboardService.anomalyDataChange.changed($scope.anomalyChartRecords || [], timerange.earliestMs, timerange.latestMs);
+
+      if (cellData !== undefined && cellData.fieldName === undefined) {
+        // Click is in one of the cells in the Overall swimlane - reload the 'view by' swimlane
+        // to show the top 'view by' values for the selected time.
+        loadViewBySwimlaneForSelectedTime(timerange.earliestMs, timerange.latestMs);
+        $scope.viewByLoadedForTimeFormatted = moment(timerange.earliestMs).format('MMMM Do YYYY, HH:mm');
```

Comment:
probably not for this PR, but we really should pull this time format string out and into a constant.
we use this and variations of it quite a few times throughout the app.

Reply:
I added a note in the Anomaly Explorer's meta issue: https://github.com/elastic/kibana/issues/18553